src/nvim/api/buffer.c

```diff
@@ -1060,7 +1140,14 @@ void nvim_buf_delete(Buffer buffer, Dict(buf_delete) *opts, Error *err)
 {
   buf_T *buf = find_buffer_by_handle(buffer, err);
 
-  if (ERROR_SET(err)) {
+  if (!buf || ERROR_SET(err)) {
+    return;
+  }
+
+  // do we care about loading the buffer if we are deleting it?
+  // load buffer first if it's not loaded
+  if (!buf_ensure_loaded(buf)) {
```

Comment:
would it be okay if we don't try to load the buffer if we are deleting it anyway? not so sure

Reply:
yep, it's redundant in this case.